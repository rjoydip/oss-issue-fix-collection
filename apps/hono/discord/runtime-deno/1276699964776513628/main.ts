import { Env, Hono, Input, MiddlewareHandler, TypedResponse } from "hono";
import { createMiddleware } from "hono/factory";

interface Response<T> {
  message: string;
  data: T;
}

const app = new Hono<Env>();

const authMiddleware = <
  E extends Env,
  P extends string,
  I extends Input,
>(message: string = "Hello!"): MiddlewareHandler<E, P, I> => {
  return createMiddleware(async (c, next) => {
    await next();
    c.res.headers.append("X-Message", message);
    c.body("");
  });
};

app.get("/", authMiddleware(), (c) => {
  return c.json({
    message: "Success",
    data: "Hello Hono!",
  });
});

app.post("/", (c): TypedResponse<Response<string>, 200> => {
  return c.json({
    message: "Success",
    data: "Hello Hono!",
  });
});

Deno.serve(app.fetch);
