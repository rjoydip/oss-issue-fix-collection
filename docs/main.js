import LumeCode from "./scripts/components/lume_code.js";
import LumeCarouselControls from "./scripts/components/lume_carousel_controls.js";
import LumeFilter from "./scripts/components/lume_filter.js";
import LumeDevices from "./scripts/components/lume_devices.js";
import LumeShield from "./scripts/components/lume_shield.js";
import LumeCopy from "./scripts/components/lume_copy.js";

customElements.define("lume-code", LumeCode);
customElements.define("lume-carousel-controls", LumeCarouselControls);
customElements.define("lume-filter", LumeFilter);
customElements.define("lume-devices", LumeDevices);
customElements.define("lume-shield", LumeShield);
customElements.define("lume-copy", LumeCopy);

// For testing purpose of CSP middleware
const userAgentString = navigator.userAgent;
const chromeAgent = userAgentString.indexOf("Chrome") > -1;

if (chromeAgent) {
  const observer = new ReportingObserver((reports) => {
    for (const report of reports) {
      console.log(report.type, report.url, report.body);
    }
  }, { buffered: true });

  observer.observe();
}
