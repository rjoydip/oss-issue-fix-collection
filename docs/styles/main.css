@import "styles/components/button.css";
@import "styles/components/selector.css";
@import "styles/components/tag.css";
@import "styles/components/navbar.css";
@import "styles/components/footer.css";
@import "styles/components/code.css";
@import "styles/components/terminal.css";
@import "styles/components/toc.css";
@import "styles/components/site.css";
@import "styles/components/tip.css";
@import "styles/components/type.css";
@import "styles/components/device.css";
@import "styles/components/shield.css";
