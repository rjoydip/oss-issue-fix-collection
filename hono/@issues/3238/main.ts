import { Hono } from "hono";
import { serveStatic } from "hono/deno";

const app = new Hono();

app.get("/", (c) => c.text("Hello Deno!"));

app.use(
  "/*",
  serveStatic({
    root: "./static",
  }),
);

Deno.serve({ port: 3000 }, app.fetch);
