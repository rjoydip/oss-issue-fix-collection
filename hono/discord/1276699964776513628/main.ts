import { Hono, TypedResponse } from 'hono'

interface Response<T> {
  message: string;
  data: T;
}

const app = new Hono()

app.get('/', (c): TypedResponse<Response<string>, 200> => {
  return c.json({
    message: 'Success',
    data: 'Hello Hono!'
  })
})

Deno.serve(app.fetch)
