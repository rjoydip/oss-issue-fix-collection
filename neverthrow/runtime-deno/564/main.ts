import { fromPromise } from "npm:neverthrow"

export function isPromiseAllSettledOK<T>(promises: Promise<T>[]) {
  return fromPromise(Promise.allSettled([...promises]), () => new Error('Promise allSettled error'))
}

export function isPromiseAllOK<T>(promises: Promise<T>[]) {
  return fromPromise(Promise.all([...promises]), () => new Error('Promise all error'))
}