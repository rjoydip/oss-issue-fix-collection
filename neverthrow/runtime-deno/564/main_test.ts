import { assertSnapshot } from "@std/testing/snapshot";
import { isPromiseAllOK, isPromiseAllSettledOK } from "./main.ts";

Deno.test("testIsPromiseAllSettledOK", async function (t):  Promise<void> {
  const result = await isPromiseAllSettledOK<string | number>([
    Promise.resolve("Success"),
    Promise.reject("Error"),
    Promise.resolve(1),
  ])
    
  await assertSnapshot(t, result);
});

Deno.test("testIsPromiseAllOK", async function (t):  Promise<void> {
  const result = await isPromiseAllOK<string | number>([
    Promise.resolve("Success"),
    Promise.reject("Error"),
    Promise.resolve(1),
  ])
    
  await assertSnapshot(t, result);
});