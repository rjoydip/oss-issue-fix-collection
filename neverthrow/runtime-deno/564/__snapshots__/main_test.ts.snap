export const snapshot = {};

snapshot[`testIsPromiseAllSettledOK 1`] = `
Ok {
  value: [
    {
      status: "fulfilled",
      value: "Success",
    },
    {
      reason: "Error",
      status: "rejected",
    },
    {
      status: "fulfilled",
      value: 1,
    },
  ],
}
`;

snapshot[`testIsPromiseAllOK 1`] = `
Err {
  error: Error: Promise all error
    at file:///C:/Users/rjoydip/codebase/oss-issue-solving/neverthrow/runtime-deno/564/main.ts:8:56
    at file:///C:/Users/rjoydip/AppData/Local/deno/npm/registry.npmjs.org/neverthrow/7.0.1/dist/index.cjs.js:100:35
    at async file:///C:/Users/rjoydip/codebase/oss-issue-solving/neverthrow/runtime-deno/564/main_test.ts:15:18
    at async innerWrapped (ext:cli/40_test.js:191:5)
    at async exitSanitizer (ext:cli/40_test.js:107:27)
    at async outerWrapped (ext:cli/40_test.js:134:14),
}
`;
